packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -300,14 +300,27 @@ export class Cluster extends Resource implements ICluster {
       ? options.type
       : cloudmap.NamespaceType.DNS_PRIVATE;
 
-    const sdNamespace = namespaceType === cloudmap.NamespaceType.DNS_PRIVATE ?
-      new cloudmap.PrivateDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
-        name: options.name,
-        vpc: this.vpc,
-      }) :
-      new cloudmap.PublicDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
-        name: options.name,
-      });
+    let sdNamespace;
+    switch (namespaceType) {
+      case cloudmap.NamespaceType.DNS_PRIVATE:
+        sdNamespace = new cloudmap.PrivateDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+          vpc: this.vpc,
+        });
+        break;
+      case cloudmap.NamespaceType.DNS_PUBLIC:
+        sdNamespace = new cloudmap.PublicDnsNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+        });
+        break;
+      case cloudmap.NamespaceType.HTTP:
+        sdNamespace = new cloudmap.HttpNamespace(this, 'DefaultServiceDiscoveryNamespace', {
+          name: options.name,
+        });
+        break;
+      default:
+        throw new Error('Namespace type is not valid/supported.');
```

Comment:
Is there any way we can test for this? If for example the NameSpaceType enum doesn't allow custom values then we can skip adding a unit test, but otherwise we should add a test.

Reply:
NameSpaceType enum is not allowing custom values and we can cannot add test case for this.